crates/worker/src/clients/dynamodb_clients/client.rs

```diff
@@ -0,0 +1,247 @@
+    ) -> Result<(), DBUpdateError> {
+        let _ = self
+            .client
+            .update_item()
+            .table_name(self.table_name.clone())
+            .key(Item::primary_key_name(), AttributeValue::S(id.to_string()))
+            .update_expression("SET #status = :toStatus")
+            .condition_expression("#status = :fromStatus")
+            .expression_attribute_names("#status", Status::attribute_name())
+            .expression_attribute_values(":toStatus", AttributeValue::N(u32::from(to).to_string()))
+            .expression_attribute_values(
+                ":fromStatus",
+                AttributeValue::N(u32::from(from).to_string()),
+            )
+            .send()
+            .await?;
+        Ok(())
+    }
+
+    pub async fn update_item_status_conditional_attempt(
+        &self,
+        id: &str,
+        from: &Status,
+        to: &Status,
+    ) -> Result<Option<()>, DBUpdateError> {
+        match_result!(
+            DBUpdateError,
+            self.update_item_status_conditional(id, from, to).await
+        )
+    }
+
+    pub async fn get_item(&self, key: &str) -> Result<Option<Item>, DBError> {
+        let result = self
+            .client
+            .get_item()
+            .table_name(self.table_name.clone())
+            .key(Item::primary_key_name(), AttributeValue::S(key.to_string()))
+            .send()
+            .await?;
+
+        if let Some(item) = result.item {
+            Ok(Some(item.try_into()?))
+        } else {
+            Ok(None)
+        }
+    }
+
+    pub async fn get_item_attempt(&self, key: &str) -> Result<Option<Option<Item>>, DBError> {
+        match self.get_item(key).await {
+            Err(DBError::GetItemError(err)) => {
+                match_result!(DBGetError, Err(err)).map_err(DBError::from)
+            }
+            result => result.map(|value| Some(value)),
+        }
+    }
+
+    pub async fn scan_items_prior_to(
+        &self,
+        time: &DateTime<Utc>,
+        exclusive_start_key: &Option<HashMap<String, AttributeValue>>,
+    ) -> Result<ScanOutput, DBScanError> {
+        const MAX_CAPACITY: usize = 100;
+
+        self.client
+            .scan()
+            .table_name(self.table_name.clone())
+            .filter_expression("CreatedAt <= :created_at")
+            .expression_attribute_values(":created_at", AttributeValue::S(time.to_rfc3339()))
+            .limit(MAX_CAPACITY as i32)
+            .set_exclusive_start_key(exclusive_start_key.clone())
+            .send()
+            .await
+    }
+
+    pub async fn scan_items_prior_to_attempt(
+        &self,
+        time: &DateTime<Utc>,
+        exclusive_start_key: &Option<HashMap<String, AttributeValue>>,
+    ) -> Result<Option<ScanOutput>, DBScanError> {
+        match_result!(
+            DBScanError,
+            self.scan_items_prior_to(time, exclusive_start_key).await
+        )
+    }
+}
+
+#[derive(Default)]
+pub enum DynamoDBAction {
+    #[default]
+    Default,
```

Comment:
Just curious: why do we need the default field here?

Reply:
Because of borrow checker in Retrier::resend_pending_actions algorithm, which is pretty much Vec::retain. Because. We need to take ownership of action to extract oneshot::sender, but we can't since its within an array we reference, so we replace is with Default. 

All defaults at the end will wether be overridden or truncated